GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -332,19 +333,20 @@ private static bool CheckRowCacheIsDirty(IList<RevisionGraphRow> orderedRowCache
             }
 
             int indexToCompare = orderedRowCache.Count - 1;
-            return orderedRowCache[indexToCompare].Revision != orderedNodesCache[indexToCompare];
+            return orderedRowCache[indexToCompare].Revision != orderedNodesCache[indexToCompare]
+                || orderedRowCache[0].Revision != orderedNodesCache[0];
         }
 
-        private void BuildOrderedRowCache(RevisionGraphRevision[] orderedNodesCache, int currentRowIndex, int lastToCacheRowIndex)
+        private void BuildOrderedRowCache(RevisionGraphRevision[] orderedNodesCache, int lastToCacheRowIndex)
         {
             int lastOrderedNodeIndex = orderedNodesCache.Length - 1;
 
             // Ensure we keep using the same instance of the rowcache from here on
             IList<RevisionGraphRow>? localOrderedRowCache = _orderedRowCache;
 
-            if (localOrderedRowCache is null || CheckRowCacheIsDirty(localOrderedRowCache, orderedNodesCache))
+            if (localOrderedRowCache is null || IsRowCacheDirty(localOrderedRowCache, orderedNodesCache))
             {
-                localOrderedRowCache = new List<RevisionGraphRow>(capacity: Math.Max(currentRowIndex, lastOrderedNodeIndex) + 1);
```

Comment:
So currentRowIndex is always smaller than lastOrderedNodeIndex?

Reply:
No, this was a wrong shortcut. Reverted.